src/ddmd/mtype.d

```diff
@@ -2717,6 +2717,23 @@ extern (C++) abstract class Type : RootObject
      */
     final Expression noMember(Scope* sc, Expression e, Identifier ident, int flag)
     {
+        //printf("Type.noMember(e: %s ident: %s flag: %d)\n", e.toChars(), ident.toChars(), flag);
+
+        static __gshared int nest;      // https://issues.dlang.org/show_bug.cgi?id=17380
+
+        static Expression returnExp(Expression e)
+        {
+            --nest;
```

Comment:
What better use-case could there be for `scope (exit)`? This function should also be outside any hot code path, if performance was a criteria to not use `scope (exit)` (though we should really fix any such issues).